core/domain/stats_domain.py

```diff
@@ -1351,3 +1365,312 @@ def validate(self):
+                raise utils.ValidationError(
+                    'For entity type exploration, the state reference '
+                    'should be of the form \'exp_id.state_name\', but '
+                    'received %s' % (self.state_reference))
+        elif self.entity_type == feconf.ENTITY_TYPE_QUESTION:
+            if len(split_state_reference) != 1:
+                raise utils.ValidationError(
+                    'For entity type question, the state reference should '
+                    'be of the form \'question_id\', but received %s' % (
+                        self.state_reference))
+        else:
+            raise utils.ValidationError(
+                'Invalid entity type received %s' % (self.entity_type))
+
+        if not isinstance(self.interaction_id, basestring):
+            raise utils.ValidationError(
+                'Expected interaction_id to be a string, received %s' % str(
+                    self.interaction_id))
+
+        if (self.interaction_id not in
+                interaction_registry.Registry.get_all_interaction_ids()):
+            raise utils.ValidationError(
+                'Unknown interaction_id: %s' % self.interaction_id)
+
+        if self.interaction_id in feconf.INTERACTION_IDS_WITHOUT_ANSWER_DETAILS:
+            raise utils.ValidationError(
+                'The %s interaction does not support soliciting '
+                'answer details from learners.' % (self.interaction_id))
+
+        if not isinstance(self.learner_answer_info_list, list):
+            raise utils.ValidationError(
+                'Expected learner_answer_info_list to be a list, '
+                'received %s' % str(self.learner_answer_info_list))
+
+        for learner_answer_info in self.learner_answer_info_list:
+            learner_answer_info.validate()
+
+        if not isinstance(self.learner_answer_info_schema_version, int):
+            raise utils.ValidationError(
+                'Expected learner_answer_info_schema_version to be an int, '
+                'received %s' % self.learner_answer_info_schema_version)
+
+        if not isinstance(self.accumulated_answer_info_json_size_bytes, int):
+            raise utils.ValidationError(
+                'Expected accumulated_answer_info_json_size_bytes to be an int '
+                'received %s' % self.accumulated_answer_info_json_size_bytes)
+
+    def add_learner_answer_info(self, learner_answer_info):
+        """Adds new learner answer info in the learner_answer_info_list.
+
+        Args:
+            learner_answer_info: LearnerAnswerInfo. The learner answer info
+                object, which is created after the learner has submitted the
+                details of the answer.
+        """
+        learner_answer_info_dict_size = (
+            learner_answer_info.get_learner_answer_info_dict_size())
+        if (self.accumulated_answer_info_json_size_bytes +
+                learner_answer_info_dict_size <= (
```

Comment:
We need a test for "big" answers that cause the limit to be exceeded. I don't see such a test in the test file, could you please add it?

Reply:
Added

Reply:
Done